### PR TITLE
Enable Gradle Workflow on Windows

### LINKED ISSUE #8072: Gradle Workflow doesn't work on Windows
```shell
C:\src> flutter create myFirstGradleAppOnWindows
Creating project myFirstGradleAppOnWindows...
C:\src> cd .\myFirstGradleAppOnWindows\
C:\src\myFirstGradleAppOnWindows> flutter run --release
Launching lib/main.dart on Nexus 6P in release mode...
Running 'gradle wrapper'...
Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.
java.lang.IllegalStateException: buildToolsVersion is not specified.
        at com.google.common.base.Preconditions.checkState(Preconditions.java:173)
        at com.android.build.gradle.BasePlugin.createAndroidTasks(BasePlugin.java:645)
        at com.android.build.gradle.BasePlugin$10.call(BasePlugin.java:608)
        at com.android.build.gradle.BasePlugin$10.call(BasePlugin.java:605)
        at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:156)
        at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:120)
        at com.android.build.gradle.BasePlugin.lambda$createTasks$1(BasePlugin.java:603)
        at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:93)
        at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:82)
        at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:44)
        at org.gradle.internal.event.BroadcastDispatch.dispatch(BroadcastDispatch.java:79)
        at org.gradle.internal.event.BroadcastDispatch.dispatch(BroadcastDispatch.java:30)
        at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
        at com.sun.proxy.$Proxy15.afterEvaluate(Unknown Source)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.notifyAfterEvaluate(LifecycleProjectEvaluator.java:67)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.evaluate(LifecycleProjectEvaluator.java:61)
        at org.gradle.api.internal.project.DefaultProject.evaluate(DefaultProject.java:573)
        at org.gradle.api.internal.project.DefaultProject.evaluate(DefaultProject.java:125)
        at org.gradle.execution.TaskPathProjectEvaluator.configureHierarchy(TaskPathProjectEvaluator.java:47)
        at org.gradle.configuration.DefaultBuildConfigurer.configure(DefaultBuildConfigurer.java:38)
        at org.gradle.initialization.DefaultGradleLauncher$2.run(DefaultGradleLauncher.java:151)
        at org.gradle.internal.Factories$1.create(Factories.java:22)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:53)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:148)
        at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:33)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:112)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:106)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:63)
        at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:106)
        at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:92)
        at org.gradle.launcher.exec.GradleBuildController.run(GradleBuildController.java:66)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:41)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:79)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:51)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:59)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:47)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.HintGCAfterBuild.execute(HintGCAfterBuild.java:44)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:293)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
        at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\src\myFirstGradleAppOnWindows\android\app\build.gradle' line: 15

* What went wrong:
A problem occurred evaluating project ':app'.
> java.net.URISyntaxException: Illegal character in opaque part at index 5: C:srclutter/packages/flutter_tools/gradle/flutter.gradle

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Exception from flutter run: Exception: Exception: Gradle failed: 1
package:flutter_tools/src/base/common.dart 30              throwToolExit
package:flutter_tools/src/android/gradle.dart 175          ensureGradlew.<async>
dart:async                                                 _SyncCompleter.complete
package:flutter_tools/src/base/process.dart 113            runCommandAndStreamOutput.<async>
===== asynchronous gap ===========================
dart:async                                                 _Completer.completeError
package:flutter_tools/src/android/gradle.dart 186          ensureGradlew.<async>
dart:async                                                 _SyncCompleter.complete
package:flutter_tools/src/base/process.dart 113            runCommandAndStreamOutput.<async>
===== asynchronous gap ===========================
dart:async                                                 _asyncThenWrapperHelper
package:flutter_tools/src/android/gradle.dart              ensureGradlew
package:flutter_tools/src/android/gradle.dart 204          buildGradleProject.<async>
===== asynchronous gap ===========================
dart:async                                                 Future.Future.microtask
package:flutter_tools/src/android/gradle.dart              buildGradleProject
package:flutter_tools/src/commands/build_apk.dart 612      buildAndroidWithGradle.<async>
===== asynchronous gap ===========================
dart:async                                                 Future.Future.microtask
package:flutter_tools/src/commands/build_apk.dart          buildAndroidWithGradle
package:flutter_tools/src/commands/build_apk.dart 622      buildApk.<async>
===== asynchronous gap ===========================
dart:async                                                 Future.Future.microtask
package:flutter_tools/src/commands/build_apk.dart          buildApk
package:flutter_tools/src/android/android_device.dart 287  AndroidDevice.startApp.<async>
dart:async                                                 _SyncCompleter.complete
package:flutter_tools/src/base/process.dart 113            runCommandAndStreamOutput.<async>
===== asynchronous gap ===========================
dart:async                                                 _asyncThenWrapperHelper
package:flutter_tools/src/android/android_device.dart      AndroidDevice.startApp
package:flutter_tools/src/run_cold.dart 105                ColdRunner._run.<async>
dart:async                                                 _SyncCompleter.complete
package:flutter_tools/src/resident_runner.dart 185         ResidentRunner.startEchoingDeviceLog.<async>
===== asynchronous gap ===========================
dart:async                                                 _asyncThenWrapperHelper
package:flutter_tools/src/run_cold.dart                    ColdRunner._run
package:flutter_tools/src/run_cold.dart 49                 ColdRunner.run.<fn>
package:stack_trace                                        Chain.capture
package:flutter_tools/src/run_cold.dart 48                 ColdRunner.run
package:flutter_tools/src/commands/run.dart 278            RunCommand.runCommand.<async>
```

### PR DESCRIPTION
**Pull request by goderbauer**
With this, the app created by `flutter create` now compiles on Windows.

Fixes https://github.com/flutter/flutter/issues/8072

/cc @mit-mit @jakobr-google

### CODE REVIEW

**Comment by goderbauer** [comment]
This will also make the converted examples from https://github.com/flutter/flutter/pull/8173 work on Windows.

**Comment by mit-mit** [comment]
Woot, thanks so much Michael! 

I'll leave the actual review for Jakob.

File: packages/flutter_tools/gradle/flutter.gradle [hunk 1]
```diff
@@ -70,6 +71,12 @@ class FlutterPlugin implements Plugin<Project> {
             throw new GradleException("flutter.sdk must point to the Flutter SDK directory")
         }
 
+        String flutterExecutableName = localProperties.getProperty("flutter.executable")
+        if (flutterExecutableName == null) {
+            throw new GradleException("flutter.executable must be set to the name of the flutter executable")
+        }
+        flutterExecutable = Paths.get(flutterRoot.absolutePath, "bin", flutterExecutableName).toFile();
```
Comment (jakobr-google) [inline packages/flutter_tools/gradle/flutter.gradle]: I'd put the OS check here, so we don't need to add an extra property to local.properties. But either way works.

**Review by jakobr-google — APPROVED** [review]
LGTM. Thanks!

File: packages/flutter_tools/gradle/flutter.gradle (hunk 1 above)
Comment (goderbauer) [inline packages/flutter_tools/gradle/flutter.gradle]: Done. @jakobr-google  can you take another quick look? Thanks.

**Review by jakobr-google — APPROVED** [review]
Thanks!

**Pull request merged**